File: bridge-pattern/java/src/AbstractShape.java
package src;

// 定义抽象桥接类，关联具体的DrawTool操作接口
// 供具体的Shape继承，以及聚合DrawTool的实现
public abstract class AbstractShape {
   protected DrawTool drawTool;

   protected AbstractShape(DrawTool drawTool) {
      this.drawTool = drawTool;
   }

   public abstract void draw(int x, int y, int radius);
}

File: bridge-pattern/go/src/AbstractShape.go
package src

// 定义抽象的Shape接口，关联具体的DrawTool操作接口，桥接具体的Shape实现与DrawTool实现
type AbstractShape interface {
  SetDrawTool(drawTool DrawTool)
  GetDrawTool() DrawTool
  Draw(x int, y int, radius int)
}
